crates/optimism/chainspec/src/lib.rs

```diff
@@ -194,6 +199,66 @@ impl OpChainSpec {
     pub fn from_genesis(genesis: Genesis) -> Self {
         genesis.into()
     }
+
+    fn get_or_init_sealed_genesis_header(&self) -> &SealedHeader {
+        self.inner.genesis_header.get_or_init(|| {
+            let header = self.make_genesis_header();
+            if self.is_optimism_mainnet() {
+                // for OP mainnet we have to do this because the genesis header can't be
+                // properly computed from the genesis.json file due to OVM history
+                return SealedHeader::new(header, OP_MAINNET_GENESIS_HASH)
```

Comment:
this is problematic because this can also be a modified genesis file

Reply:
p sure that would make it a devnet

Reply:
what if you run a custom genesis with mainnet chainid?

we can't do that here this must be set on the static MAINNET

Reply:
ok. if that's not running on LAN then it's useful for dos:ing op mainnet 😅